ui/redux/rootReducer.js

```diff
@@ -0,0 +1,67 @@
+import { combineReducers } from 'redux'
+import { reducer as formReducer, SubmissionError } from 'redux-form'
+
+import { reducers as dashboardReducers } from 'pages/Dashboard/reducers'
+import { HttpRequestHelper } from 'shared/utils/httpRequestHelper'
+import { createObjectsByIdReducer, loadingReducer, zeroActionsReducer } from './utils/reducerFactories'
+
+/**
+ * Action creator and reducers in one file as suggested by https://github.com/erikras/ducks-modular-redux
+ */
+
+// actions
+const REQUEST_PROJECTS = 'REQUEST_PROJECTS'
+const RECEIVE_PROJECTS = 'RECEIVE_PROJECTS'
+const UPDATE_PROJECT_CATEGORIES_BY_GUID = 'UPDATE_PROJECT_CATEGORIES_BY_GUID'
+
+// action creators
+export const fetchProjects = () => {
+  return (dispatch) => {
+    dispatch({ type: REQUEST_PROJECTS })
+    new HttpRequestHelper('/api/dashboard',
+      (responseJson) => {
+        dispatch({ type: UPDATE_PROJECT_CATEGORIES_BY_GUID, updatesById: responseJson.projectCategoriesByGuid })
+        dispatch({ type: RECEIVE_PROJECTS, updatesById: responseJson.projectsByGuid })
+      },
+      e => dispatch({ type: RECEIVE_PROJECTS, error: e.message, updatesById: {} }),
+    ).get()
+  }
+}
+
+export const updateProject = (values) => {
+  return (dispatch) => {
+    const urlPath = values.projectGuid ? `/api/project/${values.projectGuid}` : '/api/project'
+    const projectField = values.projectField ? `_${values.projectField}` : ''
```

Comment:
Can this line be deleted and then just do
`${urlPath}/${action}_project_${values.projectField}` ?

Also would it be better to just return immediately if values.projectField is null?

It might be worth adding /** .. */ docs explaining what the `values` arg is.

Reply:
well most of the time you are updating  projects with no specific fields so not having a project field makes sense and you end up with urls like `create_project`, `update_project`etc 

The reason projectField exists as an option at all is for `update_project_categories` which needs to be a separate url